flocker/control/_model.py

```diff
@@ -17,27 +17,72 @@
 from characteristic import attributes
 
 from twisted.python.filepath import FilePath
+
 from pyrsistent import (
     pmap, PRecord, field, PMap, CheckedPSet, CheckedPMap,
+    optional as optional_type
     )
 
 from zope.interface import Interface, implementer
 
 
-def pset_field(klass):
+def pset_field(item_type, optional=False):
```

Comment:
`optional` is too close to "the opposite of `mandatory`" for this API to be very clear.  How about `allow_none` or something similar?